### DISCUSSION TITLE
LAB 04.01 Task 01

### DISCUSSION THREAD

**Discussion by jaayg23** [rramosp/ai4eng.v1#203] [04 - GESTIÓN DE DATOS]
Buenas tardes, quería que me ayudara ya que el código que he escogido para realizar la tarea uno no me da los valores de media esperados y no entiendo el por qué. Agradezco cualquier ayuda y adjunto el código.

`my_d01['risk']=my_d01['risk'].fillna(round(my_d01.groupby('place')['risk'].transform(lambda x: x.shift().expanding().mean()),3))`

**Comment by imEag** [comment]
Hola! te recomiendo lo siguiente con `means = np.round(my_d01.groupby('place')['risk'].mean(), 3)` puedes obtener las medias de cada ciudad y acceder a ella través de `means['nombreDeCiudad']`, luego es cuestión de hacer un `fillna` con los valores correspondientes a cada ciudad. Cuéntame como te va.

**Reply by jaayg23** [reply]
Hola @imEag, agradezco tu ayuda, te cuento que el separar las medias fue algo que ya intente, pero no sabría como remplazar estos valores dentro del dataframe por ciudad, y buscando en la documentación no hallo esa respuesta, si pudieras seguirme ayudando estaré agradecido.

**Reply by imEag** [reply]
Una forma sencilla de resolverlo es extraer en tres variables diferentes el dataframe con los datos correspondientes a cada ciudad, esto lo puedes hacer de la siguiente manera `ciudad1_column = r01[place == 'NombreCiudad1']` para cada una de las ciudades. Después de calcular el promedio, haces un `.fillna` con los datos correspondientes a cada dataframe que sacaste anteriormente para cada ciudad. Luego reagrupas los tres nuevos dataframes en un solo (puedes usar cualquier estrategia, una opción válida es aplicar el método `.append`) y finalmente utilizas el método `.sort_index()` para organizar el nuevo dataframe.

**Reply by imEag** [reply]
Espero que te sirva, para cualquier inquietud aquí estamos.

**Reply by jaayg23** [reply]
Muchísimas gracias!!!
